internal/config/profile.go

```diff
@@ -641,31 +641,42 @@ func (p *Profile) Save() error {
 	return viper.WriteConfigAs(p.Filename())
 }
 
-// MongoCLIConfigHome retrieves configHome path based used by mongoCLI.
-func MongoCLIConfigHome() (string, error) {
+// OldMongoCLIConfigHome retrieves configHome path based used by mongoCLI.
+// Deprecated: MongoCLI versions below v1.24.0 use this path.
+func OldMongoCLIConfigHome() (string, error) {
 	if home := os.Getenv("XDG_CONFIG_HOME"); home != "" {
-		return home, nil
+		return "", errors.New("not applicable")
```

Comment:
[q] Should we really return an error here?

Reply:
not anymore since we're now changing this as well. thanks! updating